src/vehicle.h

```diff
@@ -1,24 +1,18 @@
-#ifndef _PROTO_GLOBALS_H_
-#define _PROTO_GLOBALS_H_
+#ifndef _VEH_H_
+#define _VEH_H_
 
-#include "Particle.h"
 #include "settings.h"
-#include "constants.h"
-#include "Led.h"
+
+#include "DataQueue.h"
+#include "SensorCommand.h"
+#include "DispatcherBuilder.h"
 
 #include "Sensor.h"
 #include "SensorGps.h"
 #include "SensorThermo.h"
-#include "SensorEcu.h"
 #include "SensorSigStrength.h"
 #include "SensorVoltage.h"
 
-#include "SensorCommand.h"
-#include "DataQueue.h"
-#include "DispatcherBuilder.h"
-
-#define VEHICLE_NAME "proto"
-
 extern DataQueue dataQ;
 extern SensorGps gps;
```

Comment:
If we're going to combine the vehicle headers into one, we should also remove these and put them in the implementations themselves.

Reply:
gps is still required in the main file as it also does timekeeping duties, so I think we should either extern it here or create the object in main.cpp

Reply:
we need gps and dataqueue in both main.cpp and globals.cpp.  can't see a way of doing that without including a .cpp file in one or the other, or declaring them as extern

Reply:
> gps is still required in the main file as it also does timekeeping duties, so I think we should either extern it here or create the object in main.cpp

Can we make this a virtual function in the `vehicle.h` header to retrieve the time, which the vehicles can implement based on the existence of a GPS? I think it would be better to not need to rely on specific sensors, as it would make the code a lot more portable as well.



> we need gps and dataqueue in both main.cpp and globals.cpp. can't see a way of doing that without including a .cpp file in one or the other, or declaring them as extern

I think it would be better if we don't put them in main since I would consider their inclusions something that may have side effects. Since we're merging the vehicle-specific code to a single main file, we'd probably want to abstract as much as we can from the entrypoint so it's easier to isolate changes from one vehicle's code to another.

Reply:
Good idea, I'll add this to issue #48